src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -146,7 +157,7 @@ private async ValueTask<Http3ControlStream> CreateNewUnidirectionalStreamAsync<T
             var connectionContext = await Context.ConnectionFeatures.Get<IQuicCreateStreamFeature>().StartUnidirectionalStreamAsync();
             var httpConnectionContext = new HttpConnectionContext
             {
-                ConnectionId = connectionContext.ConnectionId,
+                //ConnectionId = "", TODO getting stream ID from stream that isn't started throws an exception.
```

Comment:
Didn't you just await StartUnidirectionalStreamAsync()? How is the stream not started?

Reply:
Soooo. Right now, StartUnidirectionalStreamAsync doesn't actually start the stream, it only opens it. Until the first write happens, the streamId isn't assigned yet, and as there is no StartAsync for QuicStream at the moment, getting the connectionId throws an exception.

@scalablecory has hit this a lot as well, we are thinking of better solutions.

Reply:
Is there an issue tracking this?

Reply:
https://github.com/dotnet/runtime/pull/32143.